pages/deployments/oidc/oidc_with_aws.md

```diff
@@ -0,0 +1,168 @@
+---
+keywords: oidc, authentication, IAM, roles, AWS
+---
+
+# OIDC with AWS
+
+The [Buildkite Agent's oidc command](/docs/agent/v3/cli-oidc) allows you to request an OIDC token representing the current job. These tokens can be exchanged on federated systems like AWS for an identity and access management (IAM) role with AWS-scoped permissions.
+
+See the [OpenID Connect Core documentation](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) for more information about how OIDC tokens are constructed and how to extract and use claims.
+
+See the [AWS OpenID Connect identity provider in IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html) for more information on their implementation of the federated system.
+
+We will be using two Buildkite Plugins to utilise OIDC with AWS and our Pipelines.
+
+- Buildkite Plugin [aws-assume-role-with-web-identity-buildkite-plugin](https://github.com/buildkite-plugins/aws-assume-role-with-web-identity-buildkite-plugin)
+- Buildkite Plugin [aws-ssm-buildkite-plugin](https://github.com/buildkite-plugins/aws-ssm-buildkite-plugin)
+
+## Step 1: Setup OIDC provider in AWS account
+
+First we'll set up an AWS IAM OpenID Connect (OIDC) provider using the following configuration.
+
+  URL: `https://agent.buildkite.com`
+
+  Audience: `sts.amazonaws.com`
```

Comment:
Originally I had a screenshot in here, but I think we could instead link to the AWS documentation on _how_ to create an AWS IAM OpenID Connect provider https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html

Reply:
For tasks that are conducted on 3rd party systems, it's usually good/best practice to link through to the relevant documentation for those systems, since these 3rd parties are in control of developing these features and updating their documentation for it.

If we (re-)document such 3rd-party content in our docs, we then become responsible for updating this content whenever these 3rd parties update their product functionality.

The exception to this would be where specific steps might require specific tasks or Buildkite-specific code configurations to be specified. Therefore, it'd be better to reference the 3rd party process and make reference in our docs to things like, for example: "At step 3, copy and paste the following code snippet and modify accordingly, etc."

Reply:
Sweet, thanks, then that URL and Audience information is the only part they need to copy and paste, and we should link to that AWS documentation page.